cmake/FindCompiler.cmake

```diff
@@ -33,13 +33,13 @@ if(is_dpcpp)
   if(UNIX)
     set(UNIX_INTERFACE_COMPILE_OPTIONS -fsycl)
     set(UNIX_INTERFACE_LINK_OPTIONS -fsycl)
-    if(ENABLE_CURAND_BACKEND)
+    if(ENABLE_CURAND_BACKEND OR ENABLE_CUBLAS_BACKEND)
```

Comment:
Why do we need this addition?

Reply:
I added this because I wanted to have the `-fsycl-targets=nvptx64-nvidia-cuda-sycldevice` and `-fsycl-unnamed-lambda` flags defined when targeting the cublas backend and compiling with dpc++. I don't see anywhere else where these flags could be added to the link and the compile line.

Reply:
Right, I see that. My question was why do we need those flags for cuBLAS backend now? I am curious why we did not need them before and need them now.

Reply:
Right I will need to look into why this change was necessary. I don't think that this change belongs to this PR necessarily, so I'll remove it from here and submit a new PR with a more detailed description of the issue at hand

Reply:
Rebased to develop, and removed `ENABLE_CUBLAS_BACKEND` from this PR. I will open a separate PR with a more detailed explanation of this change